src/Microsoft.Tye.Extensions/Elastic/ElasticStackExtension.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Tye.Extensions.Elastic
+{
+    internal sealed class ElasticStackExtension : Extension
+    {
+        public override Task ProcessAsync(ExtensionContext context, ExtensionConfiguration config)
+        {
+            if (context.Application.Services.Any(s => s.Name == "elastic"))
+            {
+                context.Output.WriteDebugLine("elastic service already configured. Skipping...");
+            }
+            else
+            {
+                context.Output.WriteDebugLine("Injecting elastic service...");
+                
+                // We're using an "all-in-one" docker image for local dev that makes that
+                // easy to set up.
+                //
+                // See: https://elk-docker.readthedocs.io/
+                var elastic = new ContainerServiceBuilder("elastic", "sebp/elk")
+                {
+                    Bindings =
+                    {
+                        new BindingBuilder()
+                        {
+                            Name = "kibbana",
+                            Port = 5601,
+                            ContainerPort = 5601,
+                            Protocol = "http",
+                        },
+                        new BindingBuilder()
+                        {
+                            Name = "elastic",
+                            Port = 9200,
+                            ContainerPort = 9200,
+                            Protocol = "http",
+                        },
+                    },
+                };
+                context.Application.Services.Add(elastic);
+
+                if (config.Data.TryGetValue("logPath", out var obj) && 
+                    obj is string logPath &&
+                    !string.IsNullOrEmpty(logPath))
```

Comment:
Should we create the logPath folder if it doesn't exist, or will something downstream create the folder?

Reply:
docker will create it, I tried